xdem/coreg.py

```diff
@@ -473,7 +447,8 @@ def fit(
         :param reference_dem: 2D array of elevation values acting reference.
         :param dem_to_be_aligned: 2D array of elevation values to be aligned.
         :param inlier_mask: Optional. 2D boolean array of areas to include in the analysis (inliers=True).
-        :param transform: Optional. Transform of the reference_dem. Mandatory in some cases.
+        :param transform: Optional. Transform of the reference_dem. Mandatory if DEM provided as array.
+        :param crs: Optional. CRS of the reference_dem. Mandatory if DEM provided as array.
```

Comment:
Yes, the CRS is mandatory only since this PR (or previous one). It's needed as you say just to check that some calculations make sense, in particular for the slope calculation for NuthKaab.
I'll see if I can easily make it optional, but only mandatory whenever it matters.

Reply:
I had a quick look and I think it would be too much effort and make the code more complex than needed to make CRS optional. 
Because most functionalities are meant to be performed on Raster instances, it makes sense that we request the minimum information needed for a Raster i.e. a transform and CRS. I can't see a reasonable case for which a CRS would not be defined, apart from tests. So I'll leave it as it is. I don't think we need to change the code for hypothetical use cases for the moment. 😇